demux/demux_playlist.c

```diff
@@ -515,7 +515,10 @@ static int parse_dir(struct pl_parser *p)
     struct stream *stream = p->real_stream;
     enum autocreate_mode autocreate = AUTO_NONE;
     p->pl->playlist_dir = NULL;
-    if (p->autocreate_playlist && p->real_stream->is_local_file && !p->real_stream->is_directory) {
+    if (p->autocreate_playlist &&
+        p->real_stream->is_local_file &&
+        !p->real_stream->is_directory &&
+        strcmp(p->real_stream->path, "-") != 0) {
```

Comment:
Why `-` is local file?

Reply:
This should be fixed, our Matroska demuxer also checks that in hope that it is "normal" file. https://github.com/mpv-player/mpv/blob/b6f0599bfd0a26c3c4c8a87eeff51eb8b35bea9a/demux/demux_mkv_timeline.c#L262-L264

Reply:
`    bool is_local_file : 1; // from the filesystem`

stdin can only be from the filesystem no?

Reply:
mpv does treat `-` as a local file which makes sense.

Reply:
> stdin can only be from the filesystem no?

`echo <somedata> | mpv -` is from filesystem?

> mpv does treat - as a local file which makes sense.

Care to explain why?

Reply:
Actually, I don't think this is intended considering how it's used here https://github.com/mpv-player/mpv/blob/master/demux/demux_mkv_timeline.c#L262

I'll fix it

Reply:
Modified the patch

Reply:
Also sorry for the schizophrenic responses, I forgot that new comments on reviews aren't loaded in unless you refresh the whole page. I made the second and third comment without seeing what was said in response.

Reply:
This would also be wrong for `fd://` which is also open here. Generally I expected it to check `S_IFREG` and maybe `S_IFLNK`. We open and do it for directory check.

Reply:
>Care to explain why?

It's a `FILE` just like anything else and handled by the same stream code. Unless there's some strong reason to change the `stream->is_local_file` abstraction to not include stdin, I would lean towards just special casing `-` like the first iteration of this patch.

Reply:
Added a check for `stream->info->local_fs` as well, we don't set `is_local_fs` for fd anymore

Reply:
> It's a FILE just like anything else and handled by the same stream code. Unless there's some strong reason to change the stream->is_local_file abstraction to not include stdin

I agree with this but the only other usage of this field was to check if we should load ordered chapter sources or not, which doesn't really make sense for `-`. (Load ordered chapters from where?)

I think it's more useful to know if something is a local file on the filesystem or not.

Reply:
> > Care to explain why?
> 
> It's a `FILE` just like anything else and handled by the same stream code. Unless there's some strong reason to change the `stream->is_local_file` abstraction to not include stdin, I would lean towards just special casing `-` like in the first iteration of this patch.

This abstraction is used in two places and both of them expect "local file on filesystem".

Reply:
>This abstraction is used in two places and both of them expect "local file on filesystem".

Fair enough.